File: Android/tests/src/jp/co/riso/android/util/ImageUtilsTest.java

package jp.co.riso.android.util;

import jp.co.riso.smartdeviceapp.view.MainActivity;
import android.graphics.Bitmap;
import android.graphics.Canvas;
import android.graphics.Rect;
import android.test.ActivityInstrumentationTestCase2;

public class ImageUtilsTest extends ActivityInstrumentationTestCase2<MainActivity> {

    public ImageUtilsTest() {
        super(MainActivity.class);
    }

    public ImageUtilsTest(Class<MainActivity> activityClass) {
        super(activityClass);
    }

    protected void setUp() throws Exception {
        super.setUp();
    }

    protected void tearDown() throws Exception {
        super.tearDown();
    }

    // ================================================================================
    // Tests - constructors
    // ================================================================================

    public void testConstructor() {
        ImageUtils utils = new ImageUtils();
        assertNotNull(utils);
    }

    // ================================================================================
    // Tests - renderBmpToCanvas
    // ================================================================================

    public void testRenderBmpToCanvas_ValidParameters() {
        try {
            Bitmap bmp = Bitmap.createBitmap(1, 1, Bitmap.Config.ARGB_8888);

            Rect rect = new Rect(0, 0, 0, 0);
            Canvas canvas = new Canvas(bmp);
            ImageUtils.renderBmpToCanvas(bmp, canvas, false);
            ImageUtils.renderBmpToCanvas(bmp, canvas, false, rect);
            ImageUtils.renderBmpToCanvas(bmp, canvas, false, 0, 0, 0, 1);
            ImageUtils.renderBmpToCanvas(bmp, canvas, false, 0, 0, 0, 1, 1);
        } catch (Exception e) {
            fail();
        }
    }

    public void testRenderBmpToCanvas_NullBitmap() {
        try {
            Bitmap bmp = Bitmap.createBitmap(1, 1, Bitmap.Config.ARGB_8888);

            Rect rect = new Rect(0, 0, 0, 0);
            Canvas canvas = new Canvas(bmp);
            ImageUtils.renderBmpToCanvas(null, canvas, false);
            ImageUtils.renderBmpToCanvas(null, canvas, false, rect);
            ImageUtils.renderBmpToCanvas(null, canvas, false, 0, 0, 0, 1);
            ImageUtils.renderBmpToCanvas(null, canvas, false, 0, 0, 0, 1, 1);
        } catch (NullPointerException e) {
            fail();
        }
    }

    public void testRenderBmpToCanvas_NullCanvas() {
        try {
            Bitmap bmp = Bitmap.createBitmap(10, 10, Bitmap.Config.ARGB_8888);

            Rect rect = new Rect(0, 0, 0, 0);
            ImageUtils.renderBmpToCanvas(bmp, null, false);
            ImageUtils.renderBmpToCanvas(bmp, null, false, rect);
            ImageUtils.renderBmpToCanvas(bmp, null, false, 0, 0, 0, 1);
            ImageUtils.renderBmpToCanvas(bmp, null, false, 0, 0, 0, 1, 1);
        } catch (Exception e) {
            fail();
        }
    }
    
    public void testRenderBmpToCanvas_NullRect() {
        try {
            Bitmap bmp = Bitmap.createBitmap(1, 1, Bitmap.Config.ARGB_8888);
            Canvas canvas = new Canvas(bmp);

            ImageUtils.renderBmpToCanvas(bmp, canvas, false, null);
        } catch (Exception e) {
            fail();
        }
    }
}

File: Android/app/src/androidTest/java/jp/co/riso/android/util/ImageUtilsTest.kt
package jp.co.riso.android.util

import android.graphics.Bitmap
import android.graphics.Canvas
import android.graphics.Rect
import jp.co.riso.android.util.ImageUtils.renderBmpToCanvas
import junit.framework.TestCase
import org.junit.Test

class ImageUtilsTest {

    // ================================================================================
    // Tests - constructors
    // ================================================================================
    @Test
    fun testConstructor() {
        val utils = ImageUtils
        TestCase.assertNotNull(utils)
    }

    // ================================================================================
    // Tests - renderBmpToCanvas
    // ================================================================================
    @Test
    fun testRenderBmpToCanvas_ValidParameters() {
        try {
            val bmp = Bitmap.createBitmap(1, 1, Bitmap.Config.ARGB_8888)
            val rect = Rect(0, 0, 0, 0)
            val canvas = Canvas(bmp)
            renderBmpToCanvas(bmp, canvas, false)
            renderBmpToCanvas(bmp, canvas, false, rect)
            renderBmpToCanvas(bmp, canvas, false, 0, 0, 0f, 1f)
            renderBmpToCanvas(bmp, canvas, false, 0, 0, 0f, 1f, 1f)
        } catch (e: Exception) {
            TestCase.fail()
        }
    }

    @Test
    fun testRenderBmpToCanvas_NullBitmap() {
        try {
            val bmp = Bitmap.createBitmap(1, 1, Bitmap.Config.ARGB_8888)
            val rect = Rect(0, 0, 0, 0)
            val canvas = Canvas(bmp)
            renderBmpToCanvas(null, canvas, false)
            renderBmpToCanvas(null, canvas, false, rect)
            renderBmpToCanvas(null, canvas, false, 0, 0, 0f, 1f)
            renderBmpToCanvas(null, canvas, false, 0, 0, 0f, 1f, 1f)
        } catch (e: NullPointerException) {
            TestCase.fail()
        }
    }

    @Test
    fun testRenderBmpToCanvas_NullCanvas() {
        try {
            val bmp = Bitmap.createBitmap(10, 10, Bitmap.Config.ARGB_8888)
            val rect = Rect(0, 0, 0, 0)
            renderBmpToCanvas(bmp, null, false)
            renderBmpToCanvas(bmp, null, false, rect)
            renderBmpToCanvas(bmp, null, false, 0, 0, 0f, 1f)
            renderBmpToCanvas(bmp, null, false, 0, 0, 0f, 1f, 1f)
        } catch (e: Exception) {
            TestCase.fail()
        }
    }

    @Test
    fun testRenderBmpToCanvas_NullRect() {
        try {
            val bmp = Bitmap.createBitmap(1, 1, Bitmap.Config.ARGB_8888)
            val canvas = Canvas(bmp)
            renderBmpToCanvas(bmp, canvas, false, null)
        } catch (e: Exception) {
            TestCase.fail()
        }
    }
}